### PR TITLE
jormungandr: use user shape on instance autocomplete

### PR DESCRIPTION
**Pull request by patochectp**

### CODE REVIEW

**Review by pbougue — APPROVED** [review]

File: source/jormungandr/jormungandr/interfaces/v1/Places.py [hunk 1]
```diff
@@ -351,6 +351,13 @@ def get(self, region=None, lon=None, lat=None):
         if args['disable_geojson']:
             g.disable_geojson = True
 
+        user = authentication.get_user(token=authentication.get_token(), abort_if_no_token=False)
+
+        shape = None
+        if user and user.shape:
+            shape = json.loads(user.shape)
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py]: I think you can remove the `shape` and directly do `args['shape'] =...`

**Review by antoine-de — COMMENTED** [review]
It's not mandatory but if it's not too bothering it might be nice to add some integration tests with a mock user providing a shape.

File: source/jormungandr/jormungandr/interfaces/v1/Places.py [hunk 2]
```diff
@@ -351,6 +351,10 @@ def get(self, region=None, lon=None, lat=None):
         if args['disable_geojson']:
             g.disable_geojson = True
 
+        user = authentication.get_user(token=authentication.get_token(), abort_if_no_token=False)
+
+        args['shape'] = json.loads(user.shape) if user and user.shape else None
```
Comment (datanel) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py:357]: why do you move this here?
it means kraken autocomplete will also receive shape in the request?

Comment (antoine-de) [reply source/jormungandr/jormungandr/interfaces/v1/Places.py:357]: yes I think it's more coherent, the different autocompletes system might not know how to handle all the parameters but it's the same as the coordinate (to boost the results near the coordinate)

Comment (pbougue) [reply source/jormungandr/jormungandr/interfaces/v1/Places.py:357]: Yep, and each autocomplete-connector manages if it passes the shape along

File: source/jormungandr/jormungandr/tests/utils_test.py [hunk 3]
```diff
@@ -63,3 +66,108 @@ def get(self, url, *args, **kwargs):
...
+        """
+        Return True if user is blocked else False
+        """
+        return self._is_blocked
+
+def geojson():
+    return '{"type": "Feature", "geometry": ' \
+           '{"type": "Point", "coordinates": [102.0, 0.5]}, "properties": {"prop0": "value0"}}'
+
+user_in_db = {
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/tests/utils_test.py]: I don't think this should be in utils it's specific to a test no ?

File: source/jormungandr/jormungandr/tests/utils_test.py [hunk 4]
```diff
@@ -63,3 +66,108 @@ def get(self, url, *args, **kwargs):
...
+        "departure_board_test": {'ALL': True},
+        "empty_routing_test": {'ALL': True}
+    },
+}
+
+
+@contextmanager
+def user_set(app, user_name):
+    """
+    add user
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/tests/utils_test.py]: I think it's worth putting a better comment

File: source/jormungandr/tests/bragi_autocomplete_tests.py [hunk 5]
```diff
@@ -263,3 +264,40 @@ def http_get(url, *args, **kwargs):
...
+                assert False
+
+            def http_post(url, *args, **kwargs):
+                json = kwargs.pop('json')
+                assert json.get('geometry')
+                return MockResponse({}, 200, '{}')
+
+            with mock.patch('requests.get', http_get) as mock_method:
+                with mock.patch('requests.post', http_post) as mock_method:
+                    self.query('v1/coverage/main_routing_test/places?q=toto&_autocomplete=bragi')
```
Comment (antoine-de) [inline source/jormungandr/tests/bragi_autocomplete_tests.py]: can you add a boolean to assert (or a custom `mock` function) that the http_post method as bee used ?

File: source/jormungandr/jormungandr/tests/utils_test.py (hunk 4 above)
Comment (pbougue) [reply source/jormungandr/jormungandr/tests/utils_test.py]: like what?
maybe remove it, but what would you say?

File: source/jormungandr/jormungandr/tests/utils_test.py (hunk 3 above)
Comment (pbougue) [reply source/jormungandr/jormungandr/tests/utils_test.py]: hmm, we can talk about it, but to me it looks like it implies a copy/paste (or I just create a fake_user_base.py containing this part of utils file)

Comment (antoine-de) [reply source/jormungandr/jormungandr/tests/utils_test.py]: in your bragi test you only need 2 users no ?

I'm not a fan of putting test specific stuff in utils, I think it makes the tests less readable

**Review by antoine-de — APPROVED** [review]

**Comment by pbougue** [comment]
Jenkins, test this please

**Pull request merged**